addons/mail/static/src/models/message/message.js

```diff
@@ -447,7 +454,23 @@ function factory(dependencies) {
          * @returns {boolean}
          */
         _computeIsEmpty() {
-            const isBodyEmpty = (
+            return (
+                this.isBodyEmpty &&
+                !this.hasAttachments &&
+                this.tracking_value_ids.length === 0 &&
+                !this.subtype_description
+            );
+        }
+
+        /**
+         * Determine if the message body is empty regardless of the editor
+         * default html content.
+         *
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsBodyEmpty() {
```

Comment:
alphabetical (before isBodyEqualSubtypeDescription, let's ignore the misplaced isCurrentUserOrGuestAuthor for now)